### PR TITLE
Add a debugging meteor outline

### PR DESCRIPTION
**Pull request by queso**
Getting started for discussion purposes. Related to #71
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/meteor/guide/pull/96%23issuecomment-157547135%22%2C%20%22https%3A//github.com/meteor/guide/pull/96%23discussion_r45140502%22%2C%20%22https%3A//github.com/meteor/guide/pull/96%23discussion_r45140589%22%2C%20%22https%3A//github.com/meteor/guide/pull/96%23issuecomment-174191996%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/96%23issuecomment-157547135%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40queso%20thanks%20for%20writing%20up%20your%20ideas%21%20I%20think%20a%20great%20next%20step%20would%20be%20getting%20some%20people%20from%20the%20community%20who%20might%20know%20about%20debugging%20involved%20to%20see%20if%20they%20have%20any%20input.%22%2C%20%22created_at%22%3A%20%222015-11-17T23%3A46%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22Looks%20great%20%40queso%2C%20I%27ve%20just%20arrived%20in%20Meteor-land%20after%20a%20lifetime%20in%20Visual%20Studio%2C%20and%20I%20know%20I%20would%20really%20appreciate%20an%20intro%20to%20debugging.%20%20%5Cr%5Cn%5Cr%5Cn%40stubailo%20I%20accept%20the%20comments%20about%20the%20debugger%20as%20the%20last%20resort%2C%20but%20for%20those%20of%20us%20used%20to%20using%20a%20debugger%20as%20part%20of%20our%20normal%20regime%20I%20know%20it%20would%20remove%20a%20barrier%20to%20entry%20to%20feel%20it%27s%20possible%20to%20use%20one%2C%20even%20if%20it%20turns%20out%20to%20ultimately%20much%20less%20necessary%20with%20Meteor.%20%20%5Cr%5Cn%5Cr%5CnI%27m%20experimenting%20with%20both%20VS%20and%20WebStorm%2C%20which%20both%20have%20inbuilt%20debugging%20features%2C%20if%20I%20get%20that%20working%20I%27ll%20write%20up%20some%20words%20on%20that.%22%2C%20%22created_at%22%3A%20%222016-01-23T14%3A56%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11335350%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tomwoodforde%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20e7fe0e204e02d7e52856add19d2ad9ca3474806a%20outlines/debugging.md%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/96%23discussion_r45140589%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%27d%20say%20that%20methods%20like%20logging%20to%20the%20console%20and%20using%20Meteor%20shell%20to%20call%20code%20are%20simpler%20than%20diving%20into%20the%20JS%20debugger%20and%20stepping%20through%20code%20%28personally%2C%20I%20only%20use%20the%20debugger%20as%20a%20last%20resort%29.%20Perhaps%20we%20should%20start%20with%20those%20before%20hitting%20the%20super%20heavy%20duty%20tool%3F%22%2C%20%22created_at%22%3A%20%222015-11-17T23%3A47%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20outlines/debugging.md%3AL1-18%22%7D%2C%20%22Pull%20e7fe0e204e02d7e52856add19d2ad9ca3474806a%20outlines/debugging.md%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/96%23discussion_r45140502%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20think%20%60meteor%20debug%60%20is%20a%20simpler%20command%3F%22%2C%20%22created_at%22%3A%20%222015-11-17T23%3A46%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20outlines/debugging.md%3AL1-18%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/96#issuecomment-157547135'>General Comment</a></b>
- <a href='https://github.com/stubailo'></a> @queso thanks for writing up your ideas! I think a great next step would be getting some people from the community who might know about debugging involved to see if they have any input.
- <a href='https://github.com/tomwoodforde'></a> Looks great @queso, I've just arrived in Meteor-land after a lifetime in Visual Studio, and I know I would really appreciate an intro to debugging.
@stubailo I accept the comments about the debugger as the last resort, but for those of us used to using a debugger as part of our normal regime I know it would remove a barrier to entry to feel it's possible to use one, even if it turns out to ultimately much less necessary with Meteor.
I'm experimenting with both VS and WebStorm, which both have inbuilt debugging features, if I get that working I'll write up some words on that.
- [ ] <a href='#crh-comment-Pull e7fe0e204e02d7e52856add19d2ad9ca3474806a outlines/debugging.md 5'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/96#discussion_r45140502'>File: outlines/debugging.md:L1-18</a></b>
- <a href='https://github.com/stubailo'></a> I think `meteor debug` is a simpler command?
- [ ] <a href='#crh-comment-Pull e7fe0e204e02d7e52856add19d2ad9ca3474806a outlines/debugging.md 1'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/96#discussion_r45140589'>File: outlines/debugging.md:L1-18</a></b>
- <a href='https://github.com/stubailo'></a> I'd say that methods like logging to the console and using Meteor shell to call code are simpler than diving into the JS debugger and stepping through code (personally, I only use the debugger as a last resort). Perhaps we should start with those before hitting the super heavy duty tool?

<a href='https://www.codereviewhub.com/meteor/guide/pull/96?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/meteor/guide/pull/96?approve=1'></a>&nbsp;<a href='https://github.com/meteor/guide/pull/96'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by stubailo** [comment]
@queso thanks for writing up your ideas! I think a great next step would be getting some people from the community who might know about debugging involved to see if they have any input.

File: outlines/debugging.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+1. Using the javascript `debugger` keyword
+1. Using client side consoles in browsers like Chrome and Safari
+  1. Mention using debugger and the downsides of having in a loop (just close console to clear past it)
+  1. Setting breakpoints in the sources screen
+1. Starting up with `meteor --debug` to do node debugging
```
Comment (stubailo) [inline outlines/debugging.md]: I think `meteor debug` is a simpler command?

File: outlines/debugging.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+1. Using the javascript `debugger` keyword
```
Comment (stubailo) [inline outlines/debugging.md]: I'd say that methods like logging to the console and using Meteor shell to call code are simpler than diving into the JS debugger and stepping through code (personally, I only use the debugger as a last resort). Perhaps we should start with those before hitting the super heavy duty tool?

**Comment by tomwoodforde** [comment]
Looks great @queso, I've just arrived in Meteor-land after a lifetime in Visual Studio, and I know I would really appreciate an intro to debugging.  

@stubailo I accept the comments about the debugger as the last resort, but for those of us used to using a debugger as part of our normal regime I know it would remove a barrier to entry to feel it's possible to use one, even if it turns out to ultimately much less necessary with Meteor.  

I'm experimenting with both VS and WebStorm, which both have inbuilt debugging features, if I get that working I'll write up some words on that.

**Comment by stubailo** [comment]
This particular article doesn't seem to be going anywhere so I'm going to close the PR until it gets started again.

**Comment by queso** [comment]
Seems silly to close this without asking for an update. Happy to finish up, just let it sit due to not understanding how to contrib without having an actual guide up until a bit again. Happy to keep pushing on this.

@stubailo: does the guide render setup support adding images or videos to an article? I feel like this topic is better served with visuals attached.

**Comment by queso** [comment]
Ah, nm, found an example of an image at least. Video could be good too, perhaps you have an idea on that?

**Comment by stubailo** [comment]
I think you can basically put any HTML element inline in the markdown, like a YouTube embed!

Can you please address the comments on the outline? I want to make sure we get that right before getting into details about the content.

**Comment by stubailo** [comment]
@queso I think it might be better to fold this stuff into this article: #312 

What do you think?

**Comment by stubailo** [comment]
Ok it's been several months with no update, closing this for real.

**Comment by queso** [comment]
Seems fine to me, I decided that my blog post (http://joshowens.me/easily-debugging-meteor-js/) and youtube video (https://www.youtube.com/watch?v=3EXNtmgf87s) should be enough for the world, doesn't make a ton of sense to rewrite my blog post just to end up here.

**Comment by stubailo** [comment]
Sounds good!

**Pull request closed without merging**